h2o-algos/src/main/java/hex/tree/TreeHandler.java

```diff
@@ -175,6 +280,116 @@ private static void append(final int[] rightChildren, final int[] leftChildren,
                 discoveredNodes, pointer, true);
     }
 
+    private static ArrayList<Integer> extractInternalIds(TreeProperties properties) {
```

Comment:
Name of the method is misleading. This method also replaces the internal node IDs (which is not a sequence with equal distances) with a continuous equidistant sequence. Which is from user's POV more usable. I suggest to add some javadoc at least.

One question: Do we really want to replace those ? What about other tree visualization tools ? Will those node ids/numbers be comparable ?

Reply:
Changed so that we don't replace but have private "normalized" versions on TreeProperties which I can use in language representation calculation